base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -141,50 +140,54 @@ impl LMDBDatabase {
         Ok(res)
     }
 
-    fn apply_db_transaction(&mut self, txn: &DbTransaction) -> Result<(), ChainStorageError> {
+    fn apply_db_transaction(&mut self, txn: DbTransaction) -> Result<(), ChainStorageError> {
+        use WriteOperation::*;
         let write_txn =
             WriteTransaction::new(self.env.clone()).map_err(|e| ChainStorageError::AccessError(e.to_string()))?;
-        for op in txn.operations.iter() {
+        for op in txn.into_operations() {
             trace!(target: LOG_TARGET, "[apply_db_transaction] WriteOperation: {}", op);
             match op {
-                WriteOperation::Insert(insert) => self.op_insert(&write_txn, insert)?,
-                WriteOperation::Delete(delete) => self.op_delete(&write_txn, delete)?,
-                WriteOperation::UpdateBlockAccumulatedData(header_hash, data) => {
-                    self.op_update_block_accumulated_data(&write_txn, header_hash, data)?;
+                Insert(insert) => self.op_insert(&write_txn, insert)?,
+                Delete(delete) => self.op_delete(&write_txn, delete)?,
+                InsertBlockBody(header_hash, body) => {
+                    self.insert_block_body(&write_txn, header_hash, body)?;
                 },
-                WriteOperation::InsertKernel {
+                InsertKernel {
```

Comment:
These ops may no longer  be necessary.  Will have to see when pruned mode is back though

Reply:
Yeah likely, for now they are the way we do atomicity. Likely we'll move to LMDB-backed atomicity